src/app/catalog/filter/filter.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { NouisliderComponent } from 'ng2-nouislider';
+
+import { Brand, Category, Color, Size, SubCategory } from '../catalog.model';
+import { CatalogService } from '../catalog.service';
+
+@Component({
+  selector: 'catalog-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class FilterComponent implements OnInit {
+  categories: Category[] = [];
+  allsubcategories: SubCategory[] = [];
+  subcategories: SubCategory[] = [];
+  brands: Brand[] = [];
+  allbrands: Brand[] = [];
+  sizes: Size[] = [];
+  allsizes: Size[] = [];
+  searchText: string = '';
+  brandSearchText: string = '';
+  sizeSearchText: string = '';
+  rangeSlider: number[] = [100, 3000];
+  colors = Color;
+  // tslint:disable-next-line: no-any
+  rangeSliderConfig: any = {
+    behaviour: 'drag',
+    connect: true,
+    keyboard: true,
+    step: 1,
+    range: {
+      min: 0,
+      max: 10000
+    },
+    pips: {
+      mode: 'count',
+      values: 5,
+      stepped: true
+    }
+  };
+
+  constructor(private catalogService: CatalogService) {
+    this.catalogService.getAllCategories().subscribe((data) => {
+      this.categories = data;
+      this.allsubcategories = data;
+      this.subcategories = this.allsubcategories;
```

Comment:
You can try something like this.categories = this.allsubcategories = this.subcategories = data;